Storage/StorageTests/Tools/StorageTypeExtensionsTests.swift

```diff
@@ -1019,4 +1019,21 @@ class StorageTypeExtensionsTests: XCTestCase {
         // Then
         XCTAssertEqual(storedGroup, addOnGroup1)
     }
+
+    func test_loadPlugins_by_siteID_and_sorted_by_name() throws {
+        // Given
+        let plugin1 = storage.insertNewObject(ofType: SitePlugin.self)
+        plugin1.name = "BBB"
+        plugin1.siteID = sampleSiteID
+
+        let plugin2 = storage.insertNewObject(ofType: SitePlugin.self)
+        plugin2.name = "AAA"
+        plugin2.siteID = sampleSiteID
+
+        // When
+        let storedPlugins = try XCTUnwrap(storage.loadPlugins(siteID: sampleSiteID))
+
+        // Then
+        XCTAssertEqual(storedPlugins, [plugin2, plugin1])
+    }
```

Comment:
Do you think it's worth it to create a `SitePlugin` with a different `siteID` so we can test that those sites are not being loaded?

Reply:
This is another great idea 💯 I think it's good to have more tests to check for possible issues. My current test may be a bit optimistic: if the `loadPlugins` function returns all the plugins in the storage, this issue would not be caught in this test. I'll add another test to make sure this does not happen.